.travis.yml

```diff
@@ -1,26 +1,47 @@
-sudo: required
-
 language: go
-
 go:
   - 1.14.x
 
-before_install:
-  - sudo apt-get update && sudo apt-get install -y apt-transport-https
-  - curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -
-  - echo "deb https://apt.kubernetes.io/ kubernetes-xenial main" | sudo tee -a /etc/apt/sources.list.d/kubernetes.list
-  - sudo apt-get update
-  - sudo apt-get install -y kubectl
-
-addons:
-  apt:
-    packages:
-      - docker-ce
+dist: xenial
+services:
+  - docker
 
-install:
-  - echo "Please don't go get"
+env:
+  global:
+    - IP=127.0.0.1
+    - USER_NAME=$(whoami)
+    - KUBECONFIG_FOLDER_PATH=/home/$USER_NAME/.kube
+    - KUBECONFIG_PATH=$KUBECONFIG_FOLDER_PATH/config
+    - KUBECONFIG=$KUBECONFIG_PATH
+    - KUBECTL_VERSION=v1.16.8
 
-script:
-  - find .
-  - bash run-e2e-test-docker-swarm.sh
-  - bash run-e2e-test-k8s.sh
+jobs:
+  include:
+    - stage: test
+      name: faas-swarm
+      env:
+        - OPENFAAS_URL=http://$IP:8080/
+      before_install:
+        - docker swarm init
+        - git clone https://github.com/openfaas/faas.git
+        - cd faas && echo "$(pwd)" && ./deploy_stack.sh --no-auth && cd ..
+        - sleep 10
+      script: make test-swarm
+      after_script:
+        - "docker swarm leave --force"
+    # stage key is omitted because it is still par of "test" and run in parallel
+    - name: faas-netes
+      env:
+        - OPENFAAS_URL=http://$IP:31112/
+      before_install:
```

Comment:
Just trying to understand this change, was this previously in a shell script?

Reply:
yes, all of this was in the `run-e2e-tests-k8s.sh` and `run-e2e-tests-docker-swarm.sh` scripts

Reply:
For testing locally on Linux, what are the instructions and how are they different now? Do we need to update README.md?